pkg/plugins/common/config-base/v1/init.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/pflag"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/internal/validation"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/config-base/v1/scaffolds"
+)
+
+var _ plugin.InitSubcommand = &initSubcommand{}
+
+type initSubcommand struct {
+	config config.Config
+
+	// For help text.
+	commandName string
+
+	// config options
+	domain          string
+	name            string
+	componentConfig bool
+}
+
+func (p *initSubcommand) UpdateMetadata(cliMeta plugin.CLIMetadata, subcmdMeta *plugin.SubcommandMetadata) {
+	p.commandName = cliMeta.CommandName
+
+	subcmdMeta.Description = `Initialize a common project including the following files:
+  - a "PROJECT" file that stores project configuration
+  - several YAML files for project deployment under the "config" directory
+`
+	subcmdMeta.Examples = fmt.Sprintf(`  # Initialize a common project with your domain and name in copyright
+  %[1]s init --plugins common/v3 --domain example.org 
+
+  # Initialize a common project defining an specific project version
+  %[1]s init --plugins common/v3 --project-version 3
+`, cliMeta.CommandName)
+}
+
+func (p *initSubcommand) BindFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&p.domain, "domain", "my.domain", "domain for groups")
+	fs.StringVar(&p.name, "project-name", "", "name of this project")
+	fs.BoolVar(&p.componentConfig, "component-config", false,
+		"create a versioned ComponentConfig file, may be 'true' or 'false'")
```

Comment:
These 3 variables are only used to be set in config, so it can be done in any plugin. Which is the reasoning to move them to this plugin and not keeping them in the previous one?

Reply:
This plugin requires that. That is common no matter the language. 
See: https://deploy-preview-2115--kubebuilder.netlify.app/plugins/creating-plugins.html#language-based-plugins

What we could make here specific would be only the componentConfig option. However, it is also common no matter the language and then, because of it, it is here.

Reply:
I am closing this one as addressed, however,  please feel free to ping me for we speak about this one if you see that its required.

Reply:
I feel like these options could be a better fit for an additional plugin (I think we shouldn't care about plugins being too small, so a plugin with only these options is fine IMO) but we can do that in a follow up.

The reason why several small plugins are not an issue is because we use the bundle concept so that the user doesn't have to write a long list of plugins, which would be the main drawback of multiple plugins (the UX).